src/bool.h

```diff
@@ -0,0 +1,19 @@
+/// @file
+/// SPDX-License-Identifier: GPL-3.0-or-later
+/// @author Simon Heybrock
+/// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+/// National Laboratory, and European Spallation Source ERIC.
+#ifndef BOOL_H
+#define BOOL_H
+
+class Bool {
+public:
+  Bool(const bool value = false) : m_value(value) {}
```

Comment:
Implicit constructor. Allows:
```
Bool b(1);
Bool b1("anything");
```
May not be a problem for what is needed, but thought I'd point it out anyway.

Reply:
I agree that that is not nice, but without an implicit constructor conversion from `bool` also becomes explicit, which makes it unwieldy to use.

Do you know a way to get both?

Reply:
Fortunately, I think for most cases of creating a variable and setting the data, you go via an initializer list or otherwise deduce common Type T, so I don't think there's a way to fool the dataset::insert into giving you a vector of bool unless you provide all inputs `{false, true, false ..}` as bool? i.e this should not be a major concern.

I did find some musing ons this http://www.wambold.com/Martin/writings/typesafe-bool.html but doesn't look particularly nice. http://www.wambold.com/Martin/writings/typesafe-bool.html

Reply:
> I don't think there's a way to fool the dataset::insert into giving you a vector of bool unless you provide all inputs {false, true, false ..} as bool?

No, it will not fool it, passing `{false, true, false, ...}` will still store it as `std::vector<Bool>`.

Reply:
Indeed, and it will not accept mixed types even if they are implicitly convertible to bool, that just won't compile.